apollo-mockserver/src/commonMain/kotlin/com/apollographql/apollo3/mockserver/MockServerExtensions.kt

```diff
@@ -47,25 +47,46 @@ fun createMultipartMixedChunkedResponse(
     responseDelayMillis: Long = 0,
     chunksDelayMillis: Long = 0,
     boundary: String = "-",
+    waitForMoreChunks: Boolean = false,
```

Comment:
That's a lot of constructor arguments. I think MockResponse is due for a little refactoring. 

We could make it take a `Source` as body:

```kotlin
class MockResponse(
    val statusCode: Int = 200,
    val body: Source = Buffer(),
    val headers: Map<String, String> = emptyMap(),
)
```

Then `delay` and `chunks` "just" become specific `Source` implementations:

```kotlin
class DelayedSource(
    val delayMillis: Long,
    val wrappedSource:Source
    ): Source{
  private var firstRead = true

  override fun close() {
    wrappedSource.close()
  }

  override fun read(sink: Buffer, byteCount: Long): Long {
    if (firstRead) {
      // TODO: see if we can implement this more efficiently with timeouts
      usleep(delayMillis*1000)
    }
    return wrappedSource.read(sink, byteCount)
  }

  override fun timeout(): Timeout {
    TODO("Not yet implemented")
  }
}

class ChunkedSource(): Source {
  // ...
}
```

Reply:
Could we do some kind of non-blocking/polling I/O with okio? have `read()` return zero in the source implementation while waiting and something >0 when data is available again?

Reply:
So I think 0 would make callers call again immediately, in a loop, but we could sleep for a few ms to not hog the cpu? Not ideal but maybe enough for tests?

Reply:
Right. `okio.Source` is not the good abstraction then. But we could write our own `Source` abstraction with coroutines support to give a chance to JS land to "sleep"/"delay"

Reply:
What about `Flow<ByteString>`?

```
class MockResponse(
    val statusCode: Int = 200,
    val body: Flow<ByteString> = emptyFlow(),
    val headers: Map<String, String> = emptyMap(),
)
```

Example usage:

```
fun MockServer.enqueue(string: String, delayMs: Long = 0) {
  enqueue(MockResponse(
      statusCode = statusCode,
      body = flow {
        delay(delayMillis)
        emit(body.encodeUtf8())
      },
      headers = mapOf("Content-Length" to string.length.toString()),
  ))
}
```

```
val chunks = Channel<String>(Channel.UNLIMITED)
delay(200)
chunks.send("chunk 1")
delay(200)
chunks.send("chunk 2")
chunks.close()
val body: Flow<ByteString> = chunks.receiveAsFlow().map { it.encodeUtf8() }
```

Reply:
If you're OK with it I can remove the MockServer changes from this PR and open a dedicated one 😅

Reply:
Sure thing!